docs/source/reservations/reservations.rst

```diff
@@ -56,15 +56,22 @@ For more details of the fields, see :ref:`config-yaml`.
 
 .. note::
 
-    If any of the fields are specified, please allow about 30 seconds for SkyPilot optimizer to retrieve the latest reservation/block status on all regions and zones from your AWS account.
+    If any of the fields are specified, SkyPilot optimizer may take around 30 seconds to retrieve the latest reservation/block status on all regions and zones from your AWS account.
 
 
 .. _utilizing-reservations:
 
 Utilizing Reservations
 ~~~~~~~~~~~~~~~~~~~~~~
 
-By specifying configurations above, SkyPilot will prioritize using any available resources in reservation/block (i.e., consider them as zero cost) whenever you launch a cluster/job.
+By specifying configurations above, SkyPilot will prioritize using any available capacity in reservation/block (i.e., consider them as zero cost) whenever you launch a cluster/job.
+
+SkyPilot will act as follows:
+
+1. Query reservations/blocks across each AWS regions and zones to find all non-occupied capacity.
+2. Calculate per-zone pricing with non-occupied capacity as zero-cost and on-demand/spot price for the rest of resources (if any is needed to satisfy the requested resources).
+3. :ref:`Automatically failover <auto-failover>` through different zones based on the per-zone pricing until the requested resources are satisfied.
```

Comment:
```suggestion
1. Query reservations/blocks across AWS regions and zones to find all available capacity. (If the task specifies specific regions or zones to use, only those are queried.)
2. For each zone, calculate its cost: any available reserved capacity is considered as zero cost, and if any on-demand/spot resource is needed to supplement the available reserved capacity to fully satisfy the request, their on-demand/spot price is included.
3. :ref:`Automatically failover <auto-failover>` through these zones in decreasing per-zone cost order until the requested resources are provisioned.
```

Reply:
Thanks! Applied the changes. `through these zones in decreasing per-zone cost order`. I changed this to `increasing` if I understand correctly